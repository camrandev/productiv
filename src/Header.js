import React from "react";
import QuoteGetter from "./QuoteGetter";

/** header
 *
 * App -> header -> QuoteFetcher
 **/
function Header() {
  return (
    <header className="container-fluid pt-4 pb-1">
      <div className="container">
        <h1>Prøductïv</h1>
        <p className="lead">The best name in todo list management.</p>
        <QuoteGetter />
      </div>
    </header>
  );
}

export default Header;
