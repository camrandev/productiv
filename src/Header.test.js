import React from "react";
import { render } from "@testing-library/react";
import Header from "./Header";

describe('Snapshot test', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("productiv Header", function () {
  it("renders without crashing", function () {
    render(<Header />);
  });

  it("contains expected title", function () {
    const result = render(<Header />);
    expect(result.queryByText("Prøductïv")).toBeInTheDocument();
  });

  it ("rendered quotes Header", function () {
    const result = render(<Header />);
    expect(result.queryByText("Click here for an inspirational quøte!")).toBeInTheDocument();
  });
});

