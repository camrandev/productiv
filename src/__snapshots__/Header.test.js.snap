// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders correctly 1`] = `
<header
  class="container-fluid pt-4 pb-1"
>
  <div
    class="container"
  >
    <h1>
      Prøductïv
    </h1>
    <p
      class="lead"
    >
      The best name in todo list management.
    </p>
    <div
      class="text-end"
    >
      <button>
        Click here for an inspirational quøte!
      </button>
    </div>
  </div>
</header>
`;
